asciidoc-confluence-publisher-client/src/main/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisher.java

```diff
@@ -187,43 +190,41 @@ private void updatePage(String contentId, String ancestorId, ConfluencePageMetad
         }
     }
 
-    private static String contentHash(String content) {
-        return sha256Hex(content);
-    }
-
     private void addAttachments(String contentId, Map<String, String> attachments) {
         attachments.forEach((attachmentFileName, attachmentPath) -> addOrUpdateAttachment(contentId, attachmentPath, attachmentFileName));
     }
 
     private void addOrUpdateAttachment(String contentId, String attachmentPath, String attachmentFileName) {
         Path absoluteAttachmentPath = absoluteAttachmentPath(attachmentPath);
+        String newContentHash = sha256Hash(fileInputStream(absoluteAttachmentPath));
```

Comment:
Maybe `attachmentHash` or `attachmentContentHash` would be better (to distinguish from the hash of the actual content).